bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoDimmerHandler.java

```diff
@@ -0,0 +1,639 @@
+                    if (value != null) {
+                        variable = "startTime";
+                        logger.trace("New startTime '{}' for device '{}' received", value, getThing().getUID());
+                        this.onValueReceived(variable, value, actionService + "1");
+                    }
+                    value = StringUtils.substringBetween(wemoCallResponse, "<endTime>", "</endTime>");
+                    if (value != null) {
+                        variable = "endTime";
+                        logger.trace("New endTime '{}' for device '{}' received", value, getThing().getUID());
+                        this.onValueReceived(variable, value, actionService + "1");
+                    }
+                    value = StringUtils.substringBetween(wemoCallResponse, "<nightMode>", "</nightMode>");
+                    if (value != null) {
+                        variable = "nightMode";
+                        logger.trace("New nightMode state '{}' for device '{}' received", value, getThing().getUID());
+                        this.onValueReceived(variable, value, actionService + "1");
+                    }
+                    value = StringUtils.substringBetween(wemoCallResponse, "<nightModeBrightness>",
+                            "</nightModeBrightness>");
+                    if (value != null) {
+                        variable = "nightModeBrightness";
+                        logger.trace("New nightModeBrightness  '{}' for device '{}' received", value,
+                                getThing().getUID());
+                        this.onValueReceived(variable, value, actionService + "1");
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Failed to get actual NightMode state for device '{}': {}", getThing().getUID(),
+                    e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    public String getWemoURL(String actionService) {
+        URL descriptorURL = service.getDescriptorURL(this);
+        String wemoURL = null;
+        if (descriptorURL != null) {
+            String deviceURL = StringUtils.substringBefore(descriptorURL.toString(), "/setup.xml");
+            wemoURL = deviceURL + "/upnp/control/" + actionService + "1";
+            return wemoURL;
+        }
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    public State getDateTimeState(String attributeValue) {
+        if (attributeValue != null) {
+            long value = 0;
+            try {
+                value = Long.parseLong(attributeValue) * 1000; // convert s to ms
+            } catch (NumberFormatException e) {
+                logger.error("Unable to parse attributeValue '{}' for device '{}'; expected long", attributeValue,
```

Comment:
Where does this exception come from? Is it an unexpected message from the device, i.e. a bug/incompatibility in the binding, which the user should report as an issue? If so, better formulate the message in a way that the user knows that this is the case.
Is it an irrecoverable error for the binding? If it only concerns a small feature, while the rest is still working, a WARN message should do.

Reply:
reduced to warn, nothing really important